pandas/core/frame.py

```diff
@@ -10598,18 +10598,23 @@ def corrwith(
                     cols = self.columns
                     ndf = self.values.transpose()
                 k = other.values
+                k_mask = ~np.isnan(k)
+                if k.dtype == "bool":
+                    k = k.astype("float")
```

Comment:
Why are these casts needed? If needed, can it be converted to int instead?

Reply:
The bool type won't be accepted by rank_1d's signature.

Reply:
If we cast it to int, then nan in bool array will be -2147483648.

Reply:
rank should accept uint8, so k = k.view(uint8)

What happens with nullable dtypes?

Reply:
`TypeError: No matching signature found`

Reply:
rank_1d is compiled for uint8_t

Reply:
Actually not, the signature for rank_1d is `iu_64_floating_t`, uint8 is not included.

https://github.com/pandas-dev/pandas/blob/e8093ba372f9adfe79439d90fe74b0b5b6dea9d6/pandas/_libs/dtypes.pxd#L39

Reply:
Oh, sorry. My code was tested on 1.4.3.. I find the signature is changed in 1.5.0. as `numeric_object_t` instead of `iu_64_floating_t`. Now it works. Thanks.

Reply:
When nullable dtypes appears, then any r in `for i, r in enumerate(ndf)` will be Object dtype, including the none-nullable dtype. Then np.isnan raise unsafe casting error. @phofl

Reply:
This worked on 1.4.4, so have to fix this too or revert as suggested in the issue

Reply:
Any reason we can't use pd.isna?

Reply:
change to pd.isna now